### PR TITLE
[main] Add custom image builder modules

### PR DESCRIPTION
**Pull request by WickramBug**
## Purpose
Added the following modules
```
Compute-Gallery
Compute-Image
```

### CODE REVIEW

File: modules/azurerm/Compute-Gallery/compute_gallery.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+# This software is the property of WSO2 LLC. and its suppliers, if any.
+# Dissemination of any information or reproduction of any material contained
+# herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+# You may not alter or remove any copyright or other notice from copies of this content.
+#
+# --------------------------------------------------------------------------------------
+
+resource "azurerm_shared_image_gallery" "compute_gallery" {
+  name                = join("_", ["sig", var.project, var.environment, var.sig_application_name])
```
Comment (ThilinaManamgoda) [inline modules/azurerm/Compute-Gallery/compute_gallery.tf:13]: Let's join with `-`. Also let's allow users to override the name `sig-${var.name}`

File: modules/azurerm/Compute-Image/variables.tf [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#
+# This software is the property of WSO2 LLC. and its suppliers, if any.
+# Dissemination of any information or reproduction of any material contained
+# herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+# You may not alter or remove any copyright or other notice from copies of this content.
+#
+# --------------------------------------------------------------------------------------
+
+variable "padding" {
+  type = string
```
Comment (ThilinaManamgoda) [inline modules/azurerm/Compute-Image/variables.tf:13]: Description is missing

File: modules/azurerm/Compute-Image/compute_image.tf [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+#
+# This software is the property of WSO2 LLC. and its suppliers, if any.
+# Dissemination of any information or reproduction of any material contained
+# herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+# You may not alter or remove any copyright or other notice from copies of this content.
+#
+# --------------------------------------------------------------------------------------
+
+resource "azurerm_shared_image" "compute_image" {
+  name                = join("-", ["si", var.project, var.environment, var.si_application_name, var.padding])
```
Comment (ThilinaManamgoda) [inline modules/azurerm/Compute-Image/compute_image.tf:13]: Also let's allow users to override the name `si-${var.name}`

**Review by ThilinaManamgoda — CHANGES_REQUESTED** [review]

File: modules/azurerm/Compute-Gallery/compute_gallery.tf (hunk 1 above)
Comment (WickramBug) [reply modules/azurerm/Compute-Gallery/compute_gallery.tf:13]: Azure compute gallery name must contain only English letters, numbers, underscores, and periods. The name cannot begin or end with underscores or periods.

**Review by ThilinaManamgoda — APPROVED** [review]

**Review by athiththan11 — APPROVED** [review]

**Pull request merged**